pkg/v1/tkg/clusterclient/clusterclient.go

```diff
@@ -1601,8 +1601,11 @@ func (c *client) ScalePacificClusterControlPlane(clusterName, namespace string,
 	if err != nil {
 		return err
 	}
-	patchPacificControlPlaneCount := fmt.Sprintf("{\"spec\":{ \"topology\":{\"controlPlane\":{\"count\": %v}}}}", controlPlaneCount)
-	err = c.PatchResource(tkcObj, clusterName, namespace, patchPacificControlPlaneCount, types.MergePatchType, nil)
+
+	payloadFormatStr := `[{"op":"replace","path":"/spec/topology/controlPlane/replicas","value":%d}]`
+	payloadBytes := fmt.Sprintf(payloadFormatStr, controlPlaneCount)
+	log.V(3).Infof("Applying TanzuKubernetesCluster controlplane replicas update patch: %s", payloadBytes)
+	err = c.PatchResource(tkcObj, clusterName, namespace, payloadBytes, types.JSONPatchType, nil)
```

Comment:
Is this covered by any tests? At the very least, there should be unit tests in the package that cover this. I didn't find any (by running the package tests with locally).

Reply:
Thank you for catching this! I added the unit tests for `ScalePacificClusterControlPlane`.